cli/cmd/plugin/unmanaged-cluster/config/config.go

```diff
@@ -78,7 +78,7 @@ type UnmanagedClusterConfig struct {
 	Provider string `yaml:"Provider"`
 	// ProviderConfiguration offers optional provider-specific configuration.
 	// The exact keys and values accepted are determined by the provider.
-	ProviderConfiguration map[string]interface{} `yaml:"ProviderConfiguration"`
+	ProviderConfiguration string `yaml:"ProviderConfiguration"`
```

Comment:
The long-term intent of this key is to offer provider-specific configuration for providers we introduce.

For example, let's say we introduce a VM provider. In this case, we might want to do things like:

```
provideringConfiguration:
    memory: 4
    cpu: 2
```

I wouldn't want to require this as a `string` representation.

If we can't represent an arbitrary config as a `map[string]interface{}` (i grok conversion code could be somewhat complex), then let's change this to `RawProviderConfiguration` and retrain provider configuration.

Reply:
I actually like this as a string. That keeps it simple and generic so each provider can decide what to do with it. So if kind expects it to be yaml, then it can parse the string as yaml. If a new provider expects json, it can parse it as json. We just provide a convenient way to pass that through. If we add an `alexa` provider, then you can pass through `ProviderConfiguration: "alexa, make me a cluster"`. ;)

Just weighing in with an opinion.

Reply:
Yeah, that perspective makes sense. However, i'm not a fan of every provider having their config be a string when it should be a struct that can be validated against as structured data.

For example, take a look at provider specific configurations that has been proposed for CAPI and the ingress API, I think they were quite wise to offer this in a structured manner.

If you wanted this with your Alex provider, you'd say:

```
providerConfiguration:
    alexaCommandString: "alexa, make ma a cluster"
```

Reply:
which makes me think, my preference would actually be:

```
providerConfiguration:
    rawKindConfig: |
        kind: Cluster
        apiVersion: kind.x-k8s.io/v1alpha4
        nodes:
        - role: control-plane
          extraPortMappings:
          - containerPort: 888
            hostPort: 888
            listenAddress: "127.0.0.1"
            protocol: TCP
        - role: worker
        - role: worker
```

We probably need to pull this out into an issue and have a design conversation.